File: Algorithms/C++/preorder_postorder_inorder_binarytree.cpp
#include<bits/stdc++.h>
using namespace std;

struct Tree{
    int data;
    struct Tree *left, *right;
    Tree(int data){
        this->data = data;
        left = right = NULL;
    }
};

void printPostOrder(struct Tree* node){
    //left to right

    if(node == NULL) return;

    printPostOrder(node->left);
    printPostOrder(node->right);

    cout<<node->data<<" ";
}

void printPreOrder(struct Tree* node){
    //top to first left side
    if(node == NULL) return;

    cout<<node->data<<" ";
    printPreOrder(node->left);
    printPreOrder(node->right);
}

void printInOrder(struct Tree* node){
    //left top right
    if(node == NULL) return;
    printInOrder(node->left);
    cout<<node->data<<" ";
    printInOrder(node->right);
}

int main(){
    struct Tree* root = new Tree(3);
    root->left = new Tree(4);
    root->right = new Tree(5);
    root->left->left = new Tree(6);
    root->left->right = new Tree(10);
    root->right->right = new Tree(50);
    root->right->left = new Tree(7);

    printPostOrder(root);
    cout<<" "<<endl;
    printPreOrder(root);
    cout<<" "<<endl;
    printInOrder(root);
}

File: Algorithms/Java/preorderpostorderinorderbinarytree.java
class node{
    int key;
    node left, right;
    public node(int item){
        key = item;
        left = right = null;
    }
}

class preorderpostorderinorderbinarytree {
    node root;
    preorderpostorderinorderbinarytree(){
        root = null;
    }
    void printPostorder(node Node){
        if(Node == null) return;

        printPostorder(Node.left);
        printPostorder(Node.right);

        System.out.print(Node.key + " ");
    }
    void printInorder(node Node){
        if(Node == null) return;

        printInorder(Node.left);
        System.out.println(Node.key + " ");
        printInorder(Node.right);
    }
    void printPreorder(node Node){
        if(Node == null) return;

        System.out.println(Node.key + " ");
        printPreorder(Node.left);
        printPreorder(Node.right);
    }

    void printPostorder(){printPostorder(root);}
    void printInorder(){printPostorder(root);}
    void printPreorder(){printPreorder(root);}

    public static void main(String[] Args){
        preorderpostorderinorderbinarytree tree = new preorderpostorderinorderbinarytree();
        tree.root = new node(3);
        tree.root.left = new node(4);
        tree.root.right = new node(5);
        tree.root.left.left = new node(6);
        tree.root.left.right = new node(7);

        tree.printPreorder();
        tree.printInorder();
        tree.printPostorder();
    }
}
